### ISSUE TITLE
Support for Organization and Folders

### ISSUE THREAD

**Issue by AlanLonguet** [padok-team/yatas-gcp#4]
It seems that right now the plugin only supports projects references, it would be nice to pass organization ids and/or folder ids for bigger projects that relies on those grouping mechanisms.

It might be harder to implement for folders, as you would need to recurse through them since you can have folders in a folder but maybe a GCP API supports recursively finding projects in a single org or folder.

**Comment by khaoticvs** [comment]
This sounds like a clean extension of the current resource model.

Before starting implementation, I’d like to clarify:
- Should organizations/folders be represented as first-class resources, or inferred from existing hierarchy?
- Is the goal to mirror GCP’s org → folder → project structure 1:1, or allow partial adoption?

Happy to take a first pass once expectations are clear.

**Comment by kimberly-WY** [comment]
Confirmed reproducible on Node 20 + macOS. Interestingly it doesn't happen if you set NODE_OPTIONS=--max-old-space-size=512.

**Comment by viktoriavikulya** [comment]
The workaround in #142 works short-term but I think the real fix needs to happen at the connection pool level.

**Comment by Katomoto** [comment]
Good catch. This is related to the RFC change in how trailing slashes are normalized. PR incoming.

**Comment by viktoriavikulya** [comment]
One thing worth trying: setting the timeout at the transport layer rather than the request level — avoids the race condition.

**Comment by Katomoto** [comment]
Confirmed reproducible on Node 20 + macOS. Interestingly it doesn't happen if you set NODE_OPTIONS=--max-old-space-size=512.

**Comment by Katomoto** [comment]
Good catch. This is related to the RFC change in how trailing slashes are normalized. PR incoming.

**Comment by Katomoto** [comment]
The workaround in #142 works short-term but I think the real fix needs to happen at the connection pool level.

**Comment by kimberly-WY** [comment]
Good catch. This is related to the RFC change in how trailing slashes are normalized. PR incoming.

**Comment by viktoriavikulya** [comment]
The workaround in #142 works short-term but I think the real fix needs to happen at the connection pool level.

**Comment by volkkov** [comment]
Hit this same issue — the root cause was the missing null check before deserialization. Wrapping in a try/catch fixed it on our end.

**Comment by Katomoto** [comment]
The workaround in #142 works short-term but I think the real fix needs to happen at the connection pool level.

**Comment by kimberly-WY** [comment]
Confirmed reproducible on Node 20 + macOS. Interestingly it doesn't happen if you set NODE_OPTIONS=--max-old-space-size=512.

**Comment by viktoriavikulya** [comment]
Confirmed reproducible on Node 20 + macOS. Interestingly it doesn't happen if you set NODE_OPTIONS=--max-old-space-size=512.
